sidekick.py

```diff
@@ -74,24 +74,46 @@
 parser.add_argument( '--ttl', action='store', type=int, default=60,
                      help='ETCD ttl for the service announcement' )
 
+parser.add_argument( '--vulcand', action='store', type=bool, default=False,
+                    help='Selector for LB')
 
-def announce_services( services, etcd_folder, etcd_client, timeout , ttl ):
+parser.add_argument( '--type', action='store', default='http',
+                    help='type for Vulcand')
+
+def announce_services( services, etcd_folder, etcd_client, timeout , ttl, vulcand):
     for key, value in services:
         logger.info( 'Health check for {}'.format( key ) )
-
-        full_key = os.path.join( etcd_folder, key )
-
         healthy = check_health( value )
 
-        try:
-            if not healthy:
-                # Remove this server from ETCD if it exists
-                etcd_client.delete( full_key )
-            else:
-                # Announce this server to ETCD
-                etcd_client.write( full_key, value['uri'], ttl=ttl )
-        except etcd.EtcdException as e:
-            logging.error( e )
+        if vulcand:
+            backend = "/vulcand/backends/"+key+"/backend"
+            server = "/vulcand/backends/"+key+"/servers/srv1"
+            frontend = "/vulcand/frontends/"+key+"/frontend"
+            try:
+                if not healthy:
+                    # Remove this server from ETCD if it exists
+                    etcd_client.delete( backend )
+                    etcd_client.delete( server )
+                    etcd_client.delete( frontend )
+                else:
+                    # Announce this server to ETCD
+                    etcd_client.write( backend, {"Type": value['type']}, ttl=ttl)
+                    etcd_client.write( server, {"URL": "http://"+str(value['ip'])+":"+str(value['port'])}, ttl=ttl)
```

Comment:
It'd be much nicer to use the python format strings instead of string concatenation like this. https://docs.python.org/3/library/stdtypes.html#str.format